src/v/kafka/server/audit_log_manager.cc

```diff
@@ -0,0 +1,569 @@
+          std::current_exception());
+        throw;
+    }
+}
+
+ss::future<> audit_client::set_auditing_permissions() {
+    /// Give permissions to create and write to the audit topic
+    security::acl_entry acl_create_entry{
+      audit_principal,
+      security::acl_host::wildcard_host(),
+      security::acl_operation::create,
+      security::acl_permission::allow};
+
+    security::acl_entry acl_write_entry{
+      audit_principal,
+      security::acl_host::wildcard_host(),
+      security::acl_operation::write,
+      security::acl_permission::allow};
+
+    security::resource_pattern audit_topic_pattern{
+      security::resource_type::topic,
+      model::kafka_audit_logging_topic,
+      security::pattern_type::literal};
+
+    /// TODO: Add rules for User:* deny to things like deleting the topic
+    /// and other unwanted behavior. User:* should be allowed to ::describe
+    /// and alter topic configs
+    co_await _controller->get_security_frontend().local().create_acls(
+      {security::acl_binding{audit_topic_pattern, acl_create_entry},
+       security::acl_binding{audit_topic_pattern, acl_write_entry}},
+      5s);
+}
+
+ss::future<> audit_client::mitigate_error(std::exception_ptr eptr) {
+    if (_gate.is_closed() || _as.abort_requested()) {
+        /// TODO: Investigate looping behavior on shutdown
+        co_return;
+    }
+    vlog(klog.trace, "mitigate_error: {}", eptr);
+    auto f = ss::now();
+    try {
+        std::rethrow_exception(eptr);
+    } catch (client::broker_error const& ex) {
+        if (
+          ex.error == error_code::sasl_authentication_failed
+          && _has_ephemeral_credentials) {
+            f = inform(ex.node_id).then([this]() { return _client.connect(); });
+        } else {
+            throw;
+        }
+    } catch (...) {
+        throw;
+    }
+    co_await std::move(f);
+}
+
+ss::future<> audit_client::inform(model::node_id id) {
+    vlog(klog.trace, "inform: {}", id);
+
+    // Inform a particular node
+    if (id != client::unknown_node_id) {
+        return do_inform(id);
+    }
+
+    // Inform all nodes
+    return seastar::parallel_for_each(
+      _controller->get_members_table().local().node_ids(),
+      [this](model::node_id id) { return do_inform(id); });
+}
+
+ss::future<> audit_client::do_inform(model::node_id id) {
+    auto& fe = _controller->get_ephemeral_credential_frontend().local();
+    auto ec = co_await fe.inform(id, audit_principal);
+    vlog(klog.info, "Informed: broker: {}, ec: {}", id, ec);
+}
+
+ss::future<> audit_client::create_internal_topic() {
+    constexpr std::string_view retain_forever = "-1";
+    constexpr std::string_view seven_days = "604800000";
+    creatable_topic audit_topic{
+      .name = model::kafka_audit_logging_topic,
+      .num_partitions = config::shard_local_cfg().audit_log_num_partitions(),
+      .replication_factor
+      = config::shard_local_cfg().audit_log_replication_factor(),
```

Comment:
question: probably a dumb one, sorry.  Having a default RF of 3 is good, but what happens if this is used on a single node cluster?  I think __schemas has an RF of 3 as well?  How does it handle that?

Reply:
The topic will fail to be created until the value is modified in the config